### DISCUSSION TITLE
Conditional Subsetting using For Loops

### DISCUSSION THREAD

**Discussion by RanggaGemilang** [teamalgoritma/community#163] [Q&A] [answered]
![edaquestions](https://user-images.githubusercontent.com/107049922/176380272-5f0d40b3-99fa-433c-96c5-87d809865c09.png)

Hi, how can I create a new dataframe with these 10 CustomerIDs as the condition? Any explanation using **for loops** technique will be preferable but other than that will do. Thank you :)

**Note**: I have changed the CustomerID's type from 'float' to 'str' hence the list of strings below.

**Comment by kevbow** [comment, marked as answer]
Hi @RanggaGemilang,

Thanks for the question.
You can use method `.isin()` to do that (it worked kind of like for loop but built in function from pandas). The code should become like this:
`new_dataframe = france[france["CustomerID"].isin(cust10)]`

Just to point out, if you already change the type of CustomerID to string, then you should not add .0 after the value. Please try this if you meet an error later.
I hope that answer your question, and if yes please mark this as answer. Thank you

**Reply by RanggaGemilang** [reply]
![edaquestions2](https://user-images.githubusercontent.com/107049922/176389904-6ca7a71a-ac99-4f64-a82f-92f59050a335.png)
Hi @kevbow,

I tried your method above (with and without .0) but the output only returns the columns without any rows. Any idea of what happened?

**Reply by kevbow** [reply]
Hi @RanggaGemilang,

It will be easier if you share the data and notebook so I can check them, even part of the data is more than fine. As of now my deduction is something related to your CustomerID writing format. Thank you.

**Reply by RanggaGemilang** [reply]
Hi @kevbow,

The raw data is online-retail.csv from [this link](https://github.com/inytss/learnpy-da/tree/master/data_input). I can't share it to you via reply since the file size is too big. Sorry

**Reply by kevbow** [reply]
Hi Rangga,

I believe you forgot to replace the new CustomerID (str type) to your old CustomerID (float type). I tried to do the exact same step that you said above and I get the result that I want here (The only difference is I was using 6 id instead of 10).
![image](https://user-images.githubusercontent.com/68848168/176408131-8120dbff-4d28-47af-b61c-9f3ef27aa9c5.png)

**Reply by RanggaGemilang** [reply]
Ahh I see, I forgot that I've changed it into 'object' instead of 'str' hence the error. All noted and thank you very much sir @kevbow :)
